interface/orders/types.php

```diff
@@ -63,230 +66,241 @@
+            if (id) { // id == 0 means top level
+                var td1 = $('#td' + id);
+                if (td1.hasClass('isExpanded')) {
+                    $('#con' + id).remove();
+                    td1.removeClass('isExpanded');
+                    swapsign(td1, '-', '+');
+                }
+                if (td1.hasClass('haskids') && !haskids) {
+                    td1.removeClass('haskids');
+                    // swapsign(td1, '+', '.');
+                    swapsign(td1, '+', '|');
+                    return;
+                }
+                if (!td1.hasClass('haskids') && haskids) {
+                    td1.addClass('haskids');
+                    // swapsign(td1, '.', '+');
+                    swapsign(td1, '|', '+');
+                }
+                if (haskids) {
+                    td1.parent().after('<tr class="outertr"><td colspan="5" id="con' + id + '" style="padding:0">Loading...</td></tr>');
+                    td1.addClass('isExpanded');
+                    swapsign(td1, '+', '-');
+                }
+            }
+            if (haskids)
+                $.getScript('types_ajax.php?id=' + id + '&order=<?php echo $order; ?>' + '&labid=<?php echo $labid; ?>');
+            else
+                recolor();
+        }
+
+        // edit a node
+        function enode(id) {
+            dlgopen('types_edit.php?parent=0&typeid=' + id, '_blank', 600, 500, false, 'Edit Procedure Type');
+        }
+
+        // add a node
+        function anode(id) {
+            dlgopen('types_edit.php?typeid=0&parent=' + id, '_blank', 600, 500, false, 'Add Procedure Type');
+        }
+
+        // call this to alternate row colors when anything changes the number of rows
+        function recolor() {
+            var i = 0;
+            $('#con0 tr').each(function (index) {
+                // skip any row that contains other rows
+                if ($(this).hasClass('outertr')) return;
+                this.className = (i++ & 1) ? "evenrow" : "oddrow";
+            });
+        }
+
+    </script>
 
 </head>
 
 <body class="body_nav">
-<center>
-
-<h3 style='margin-top:0'><?php xl('Types of Orders and Results', 'e') ?></h3>
-
-<form method='post' name='theform' action='types.php?popup=<?php echo $popup ?>&order=<?php
-echo $order;
-if (isset($_GET['formid' ])) {
-    echo '&formid='  . attr($_GET['formid']);
-}
-
-if (isset($_GET['formseq'])) {
-    echo '&formseq=' . attr($_GET['formseq']);
-}
-?>'>
-
-<table width='100%' cellspacing='0' cellpadding='0' border='0'>
- <tr class='head'>
-  <th class='col1' align='left'>&nbsp;&nbsp;<?php xl('Name', 'e') ?></th>
-  <th class='col2' align='left'><?php xl('Order', 'e') ?></th>
-  <th class='col3' align='left'><?php xl('Code', 'e') ?></th>
-  <th class='col4' align='left'><?php xl('Description', 'e') ?></th>
-  <th class='col5' align='left'>&nbsp;</th>
- </tr>
-</table>
-
-<div id="con0">
+    <h3 style='margin-top:0'><?php xlt('Types of Orders and Results') ?></h3>
```

Comment:
need an echo here, or else it will be blank

Reply:
I try to replace the xl with xl? whenever I can but missed this echo. Nice catch and thanks. Still working to improve the lookup and pickers so will leave open for awhile.